Client/Frontend/Browser/Webview/WebviewNavigationHandler.swift

```diff
@@ -0,0 +1,67 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import WebKit
+
+protocol NavigationHandler {
+    var decisionHandler: (WKNavigationActionPolicy) -> Void { get }
+    func filterDataScheme(url: URL, navigationAction: WKNavigationAction)
+}
```

Comment:
Minor nit pick. This should be WebviewNavigationHandler as NavigationHandler is too generic. The current WebviewNavigationHandler can be called WebviewNavigationHandlerImp or WebviewNavigationHandlerImplementation or if you wanted to be specific it could be WebKitWebviewNavigationHandler.

Even more minor nit pick. Web view is 2 words so the V should be capitalized.